chia/consensus/blockchain_check_conditions.py

```diff
@@ -80,7 +79,7 @@ def blockchain_assert_relative_time_exceeds(condition: ConditionWithArgs, unspen
     return None
 
 
-def blockchain_assert_announcement(condition: ConditionWithArgs, announcements: Set[bytes]) -> Optional[Err]:
+def blockchain_assert_announcement(condition: ConditionWithArgs, announcements: List[bytes32]) -> Optional[Err]:
```

Comment:
Determining membership in a set is `O(ln n)` and in a list it's `O(n)`, so using a list here isn't ideal.